sdk/azidentity/azidentity.go

```diff
@@ -80,15 +77,60 @@ func validTenantID(tenantID string) bool {
 	return match
 }
 
-// tokenEndpoint takes a given path and appends "/token" to the end of the path
-func tokenEndpoint(p string) string {
-	return path.Join(p, "/token")
+func newPipelineAdapter(opts *azcore.ClientOptions) pipelineAdapter {
+	pl := runtime.NewPipeline(component, version, runtime.PipelineOptions{}, opts)
+	return pipelineAdapter{pl: pl}
+}
+
+type pipelineAdapter struct {
+	pl runtime.Pipeline
 }
 
-// oauthPath returns the oauth path for AAD or ADFS
-func oauthPath(tenantID string) string {
-	if tenantID == "adfs" {
-		return "/oauth2"
+func (p pipelineAdapter) CloseIdleConnections() {
+	// do nothing
+}
+
+func (p pipelineAdapter) Do(r *http.Request) (*http.Response, error) {
+	req, err := runtime.NewRequest(r.Context(), r.Method, r.URL.String())
```

Comment:
It's unfortunate that we have to do this (I feel like I had to do this elsewhere too).  Maybe we need a `runtime.WrapRequest` that creates a `runtime.Request` from an existing `http.Request`.